zx/compare.go

```diff
@@ -271,9 +271,12 @@ func CompareBstring(op string, pattern zng.Bstring) (Predicate, error) {
 // RegexpComparison returns a Predicate that compares values that must
 // be a string or enum with the value's regular expression using a regex
 // match comparison based on equality or inequality based on op.
-func CompareRegexp(op, pattern string) (Predicate, error) {
-	re, err := regexp.Compile(pattern)
+func compareRegexp(op, pattern string) (Predicate, error) {
+	re, err := regexp.Compile(string(zng.UnescapeBstring([]byte(pattern))))
 	if err != nil {
+		if strings.Contains(err.Error(), "invalid UTF-8") {
+			err = fmt.Errorf("non UTF-8 sequence in regexp \"%s\"", pattern)
+		}
```

Comment:
Nit: I think the original error message is fine and changing it unnecessary.
```sh
$ cat <<EOF > main.go
package main

import (
	"fmt"
	"regexp"
)

func main() {
	_, err := regexp.Compile("\x80")
	fmt.Println(err)
}
EOF
$ go run main.go
error parsing regexp: invalid UTF-8: `�`
```

Reply:
Well I was trying to make the message match the thing the user entered, which was `\x80` not �.  But I don't feel strongly, I can revert this if you do.

Reply:
I missed that. It's a worthwhile goal, so how about doing it for every regexp/syntax.Error?
```go
if syntaxErr, ok := err.(*syntax.Error); ok {
	syntaxErr.Expr = pattern
}
```

Reply:
clever!